src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,503 @@
+
+                auto args = FormatArguments();
+                args.push(StringIds::completed_by_name_in_years_months);
+                args.push(StringIds::buffer_2039);
+                args.push<uint16_t>(scenarioInfo->completedMonths / 12);
+                args.push<uint16_t>(scenarioInfo->completedMonths % 12);
+
+                const int16_t x = 210;
+                Gfx::drawStringCentred(*dpi, x, y + 10, Colour::black, formatStringId, &args);
+            }
+
+            y += ROW_HEIGHT;
+        }
+    }
+
+    // 0x00443E9B
+    static void onMouseUp(window* self, const widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                WindowManager::close(self->type);
+                break;
+        }
+    }
+
+    // 0x00443EA6
+    static void onMouseDown(window* self, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::tab0:
+            case widx::tab1:
+            case widx::tab2:
+            case widx::tab3:
+            case widx::tab4:
+            {
+                uint8_t selectedCategory = widgetIndex - widx::tab0;
+                if (self->current_tab == selectedCategory)
+                    return;
+
+                self->current_tab = selectedCategory;
+
+                auto& config = Config::get();
+                config.scenario_selected_tab = selectedCategory;
+                Config::write();
+
+                self->object = reinterpret_cast<char*>(0xFFFFFFFF);
+                self->callOnResize();
+                self->callPrepareDraw();
+                self->initScrollWidgets();
+                self->invalidate();
+
+                initList(self);
+                break;
+            }
+        }
+    }
+
+    // 0x00443EF6
+    static void getScrollSize(window* self, uint32_t, uint16_t*, uint16_t* const scrollHeight)
+    {
+        *scrollHeight = scenariomgr::getNumScenariosByCategory(self->current_tab) * ROW_HEIGHT;
+    }
+
+    // 0x00443F32
+    static void onScrollMouseDown(window* self, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        auto numScenarios = scenariomgr::getNumScenariosByCategory(self->current_tab);
+
+        // Mouse click sound
+        Audio::playSound(Audio::sound_id::click_down, self->x + (self->width / 2));
+
+        auto index = y / ROW_HEIGHT;
+        if (index > numScenarios)
+            return;
+
+        auto* scenarioInfo = scenariomgr::getNthScenarioFromCategory(self->current_tab, index);
+        if (scenarioInfo == nullptr)
+            return;
+
+        if (isNetworked())
+        {
+            // game command 67
+        }
+        else
+        {
+            scenario::start(scenarioInfo->filename);
+        }
+    }
+
+    // 0x00443FB2
+    static void onScrollMouseOver(window* self, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        auto numScenarios = scenariomgr::getNumScenariosByCategory(self->current_tab);
```

Comment:
Wouldn't the numscenarios check be handled by `getNthScenarioFromCategory` anyway?

Reply:
Yes, but I prefer to have the check done here, too, anyway.